### PR TITLE
add CBPP-0002 to Security Category

### PR DESCRIPTION
**Pull request by lixuna**
Refs #240 

According to https://github.com/cncf/cnf-testsuite/blob/main/RATIONALE.md#to-check-if-containers-are-running-with-non-root-user-with-non-root-membership-non_root_containers, non-root best practice is part of the security category

Add CBPP-0002: Container should execute process(es) as non-root user to Security category in https://github.com/cncf/cnf-wg/blob/main/doc/best_cnf_dev.md#best-practices-for-cnf-developers

Signed-off-by: Lucina Stricko <lucina@vulk.coop>

### CODE REVIEW

File: doc/best_cnf_dev.md [hunk 1]
```diff
@@ -57,6 +57,13 @@ Management of short-lived and long lived state, including state associated with
 
 How to ensure that components are protected against security issues, including security advisories on software components, defence against attacks, and defence in depth.
 
+### CBPP-0002: Container should execute process(es) as non-root user
+
+**Description:**
+Containers have a list of their own users independent of the host system, one of which is UID 0, the root user. Containers should run processes as a user other than root which makes it easier to run the container images securely.
+
+**Reference:** [CBPP-0002](https://github.com/cncf/cnf-wg/blob/main/cbpps/0002-no-root-in-containers.md)
```
Comment (electrocucaracha) [inline doc/best_cnf_dev.md]: I think relative paths can work here
```suggestion
**Reference:** [CBPP-0002](../cbpps/0002-no-root-in-containers.md)
```

**Review by electrocucaracha — APPROVED** [review]

**Review by tomkivlin — APPROVED** [review]
lgtm

**Review by taylor — APPROVED** [review]
lgtm

**Comment by Smitholi67** [comment]
LGTM

**Pull request merged**